pkg/service/rtcservice.go

```diff
@@ -72,11 +72,23 @@ func (s *RTCService) validate(r *http.Request) (string, routing.ParticipantInit,
 	roomName := r.FormValue("room")
 	reconnectParam := r.FormValue("reconnect")
 	autoSubParam := r.FormValue("auto_subscribe")
+	publishParam := r.FormValue("publish")
 
 	if onlyName != "" {
 		roomName = onlyName
 	}
 
+	// this is new connection for existing participant -  with publish only permissions
+	if publishParam != "" {
+		// Make sure grant has CanPublish set,
+		if claims.Video.CanPublish == nil || *claims.Video.CanPublish == false {
```

Comment:
nil CanPublish means it's allowed

```suggestion
		if claims.Video.CanPublish != nil && *claims.Video.CanPublish == false {
```